x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/rollup/RollupJobIdentifierUtils.java

```diff
@@ -133,6 +163,55 @@ private static void checkDateHisto(DateHistogramAggregationBuilder source, List<
         }
     }
 
+    private static boolean isCalendarInterval(DateHistogramInterval interval) {
+        return DateHistogramAggregationBuilder.DATE_FIELD_UNITS.containsKey(interval.toString());
```

Comment:
I'm not sure we can assume that single units == fixed time for day, minute or hour.  My understanding may be wrong, but I think that if a leap second is added to the year, any calendar unit larger than second will grow to accommodate that leap second.

++ to better documentation in general, but I was waiting on #31638 so we don't have to repeat a whole bunch, just briefly describe that we work with the limits documented in that PR.

Reply:
For posterity, Jim and I chatted about this.  We both wish calendar wasn't an option since it makes the user have to know about all the edge-cases in #31638.  But there's no getting around it since that's how `date_histogram` operates.

So we're going to document the limitation heavily, strongly suggest the user only use fixed units instead of calendar, and make sure all the examples are in fixed units (e.g. `7d` instead of `1w`).